ldp/data_structures.py

```diff
@@ -336,6 +336,25 @@ def compute_advantages(self) -> None:
             # See docstring for explanation.
             # step.metadata["advantage"] = step.value - state_values[parent_id]
 
+    def remove_nonterminal_branches(self) -> TransitionTree:
+        """Creates a new tree with only branches that end in terminal states."""
```

Comment:
Can you add a comment stating failure is not considered here

Reply:
Clarified `done=True`. Technically one could write a pathological environment in which both `step.done` and `step.failed` are true, and we'd include that branch in this method. So I don't want to claim we exclude failures.